types/time.go

```diff
@@ -46,33 +50,28 @@ func (t *Time) UnmarshalJSON(data []byte) error {
 		s = s[:target] + s[target+1:]
 	}
 
-	standard, err := strconv.ParseInt(s, 10, 64)
+	// Expects a string of length 10 (seconds), 13 (milliseconds), 16 (microseconds), or 19 (nanoseconds) representing a Unix timestamp
+	switch len(s) {
+	case 12, 15, 18:
+		s += onePadding
+	case 11, 14, 17:
+		s += twoPadding
+	}
+
+	unixTS, err := strconv.ParseInt(s, 10, 64)
 	if err != nil {
 		return err
 	}
 
 	switch len(s) {
 	case 10:
-		// Seconds
-		*t = Time(time.Unix(standard, 0))
-	case 11, 12:
-		// Milliseconds: 1726104395.5 && 1726104395.56
-		*t = Time(time.UnixMilli(standard * int64(math.Pow10(13-len(s)))))
+		*t = Time(time.Unix(unixTS, 0))
 	case 13:
-		// Milliseconds
-		*t = Time(time.UnixMilli(standard))
-	case 14:
-		// MicroSeconds: 1726106210903.0
-		*t = Time(time.UnixMicro(standard * 100))
+		*t = Time(time.UnixMilli(unixTS))
 	case 16:
-		// MicroSeconds
-		*t = Time(time.UnixMicro(standard))
-	case 17:
-		// NanoSeconds: 1606292218213.4578
-		*t = Time(time.Unix(0, standard*100))
+		*t = Time(time.UnixMicro(unixTS))
 	case 19:
-		// NanoSeconds
-		*t = Time(time.Unix(0, standard))
+		*t = Time(time.Unix(0, unixTS))
 	default:
 		return fmt.Errorf("cannot unmarshal %s into Time", string(data))
```

Comment:
🗻 🦍 Pretty sure we don't need to call string when we're also using `%s`
```suggestion
		return fmt.Errorf("cannot unmarshal %s into Time", data)
```

Reply:
nice